clippy_lints/src/methods/join_absolute_paths.rs

```diff
@@ -0,0 +1,30 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::ty::is_type_diagnostic_item;
+use rustc_ast::ast::LitKind;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::LateContext;
+use rustc_span::symbol::sym::Path;
+
+use super::JOIN_ABSOLUTE_PATHS;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>, join_arg: &'tcx Expr<'tcx>) {
+    let ty = cx.typeck_results().expr_ty(expr).peel_refs();
+    if is_type_diagnostic_item(cx, ty, Path)
+        && let ExprKind::Lit(spanned) = &join_arg.kind
```

Comment:
We should call `expr_to_init` here so that `Path::join(x)` where `x` is `let x = "/a"` is linted too.